command/plugin_status_csi.go

```diff
@@ -0,0 +1,110 @@
+package command
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+)
+
+func (c *PluginStatusCommand) csiBanner() {
+	if !(c.json || len(c.template) > 0) {
+		c.Ui.Output(c.Colorize().Color("[bold]Container Storage Interface[reset]"))
+	}
+}
+
+func (c *PluginStatusCommand) csiStatus(client *api.Client, id string) int {
+	if id == "" {
+		c.csiBanner()
+		plugs, _, err := client.CSIPlugins().List(nil)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error querying CSI plugins: %s", err))
+			return 1
+		}
+
+		if len(plugs) == 0 {
+			// No output if we have no plugins
+			c.Ui.Error("No CSI plugins")
+		} else {
+			str, err := c.csiFormatPlugins(plugs)
+			if err != nil {
+				c.Ui.Error(fmt.Sprintf("Error formatting: %s", err))
+				return 1
+			}
+			c.Ui.Output(str)
+		}
+		return 0
+	}
+
+	// Lookup matched a single plugin
+	plug, _, err := client.CSIPlugins().Info(id, nil)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error querying plugin: %s", err))
+		return 1
+	}
+
+	str, err := c.csiFormatPlugin(plug)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error formatting plugin: %s", err))
+		return 1
+	}
+
+	c.Ui.Output(str)
+	return 0
+}
+
+func (c *PluginStatusCommand) csiFormatPlugins(plugs []*api.CSIPluginListStub) (string, error) {
+	// Sort the output by quota name
+	sort.Slice(plugs, func(i, j int) bool { return plugs[i].ID < plugs[j].ID })
+
+	if c.json || len(c.template) > 0 {
+		out, err := Format(c.json, c.template, plugs)
+		if err != nil {
+			return "", fmt.Errorf("format error: %v", err)
+		}
+		return out, nil
+	}
+
+	rows := make([]string, len(plugs)+1)
+	rows[0] = "ID|Controllers Healthy|Controllers Expected|Nodes Healthy|Nodes Expected"
```

Comment:
Not a show-stopped, but would the resulting format of this maybe look nicer if we did something like: `ID|Controllers Healthy/Expected|Nodes Healthy/Expected`? That would format like:

```
ID              Controllers Healthy/Expected  Nodes Healthy/Expected
my-cool-plugin                           2/2                    10/12
```

Reply:
Done in 4f77a830a, but with left-aligned numeric columns, which is not as good